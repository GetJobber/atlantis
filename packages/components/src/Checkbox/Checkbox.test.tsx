import React from "react";
import renderer from "react-test-renderer";
import { cleanup, fireEvent, render } from "@testing-library/react";
import { Checkbox } from ".";

afterEach(cleanup);

it("renders a Checkbox", () => {
  const tree = renderer
    .create(<Checkbox label="Send me spam?" name="send_me_span" value="spam" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders a disabled Checkbox", () => {
  const tree = renderer
    .create(<Checkbox label="Dont click me" disabled={true} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders each variation of checked, defaultChecked and indeterminate", () => {
  const variations = [
    { checked: true, indeterminate: true },
    { checked: false, indeterminate: true },
    { checked: true, indeterminate: false },
    { checked: false, indeterminate: false },

    { defaultChecked: true, indeterminate: true },
    { defaultChecked: false, indeterminate: true },
    { defaultChecked: true, indeterminate: false },
    { defaultChecked: false, indeterminate: false },
  ];

  variations.forEach(variation => {
    expect(
      renderer.create(<Checkbox label="Foo" {...variation} />).toJSON(),
    ).toMatchSnapshot();
  });
});

describe("Clicking the checkbox it should call the handler", () => {
  test("with a true value", () => {
    const clickHandler = jest.fn();

    const { getByLabelText } = render(
      <Checkbox checked={false} onChange={clickHandler} label="foo" />,
    );

    fireEvent.click(getByLabelText("foo"));
    expect(clickHandler).toHaveBeenCalledWith(true);
  });

  test("with a false value", () => {
    const clickHandler = jest.fn();

    const { getByLabelText } = render(
      <Checkbox checked={true} onChange={clickHandler} label="foo" />,
    );

    fireEvent.click(getByLabelText("foo"));
    expect(clickHandler).toHaveBeenCalledWith(false);
  });
});

test("should render a description when set", () => {
  const { getByText } = render(<Checkbox description="Checkers" />);
  const description = getByText("Checkers");
  expect(description).toBeInstanceOf(HTMLParagraphElement);
});
