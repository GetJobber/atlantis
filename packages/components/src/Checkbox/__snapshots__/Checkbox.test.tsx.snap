// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Checkbox 1`] = `
<div
  className="checkbox"
>
  <label
    className="wrapper"
  >
    <span
      className="checkHolder"
    >
      <input
        aria-label="Send me spam?"
        className="input"
        name="send_me_span"
        onChange={[Function]}
        type="checkbox"
        value="spam"
      />
      <span
        className="checkBox"
      >
        <svg
          className="ZDlJU6tECg7ouG-b27gya _isfJqqskSRqop8TStvGA"
          data-testid="checkmark"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            className="_2GsLyQLHv8yNULHeXGdver"
            d="M200.655 513.229c-3.936-4.075-8.644-7.326-13.849-9.562s-10.804-3.413-16.47-3.465c-5.665-0.047-11.283 1.033-16.527 3.179-5.244 2.142-10.007 5.312-14.013 9.318s-7.174 8.768-9.32 14.012c-2.145 5.244-3.225 10.863-3.176 16.529s1.226 11.264 3.462 16.469c2.236 5.205 5.487 9.911 9.562 13.85l200.491 225.83c8.001 7.996 18.851 12.489 30.165 12.489s22.164-4.493 30.165-12.489l481.828-481.832c4.079-3.936 7.326-8.644 9.566-13.849 2.236-5.206 3.409-10.804 3.46-16.47s-1.028-11.283-3.174-16.527c-2.146-5.244-5.316-10.007-9.323-14.013-4.002-4.006-8.768-7.174-14.012-9.32s-10.863-3.225-16.525-3.176c-5.666 0.049-11.264 1.226-16.469 3.462-5.21 2.236-9.916 5.487-13.85 9.562l-451.667 451.666-170.325-195.665z"
          />
        </svg>
      </span>
    </span>
    <span
      className="label"
    >
      <p
        className="base regular base greyBlueDark"
      >
        Send me spam?
      </p>
    </span>
  </label>
</div>
`;

exports[`renders a disabled Checkbox 1`] = `
<div
  className="checkbox"
>
  <label
    className="wrapper disabled"
  >
    <span
      className="checkHolder"
    >
      <input
        aria-label="Dont click me"
        className="input"
        disabled={true}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="checkBox"
      >
        <svg
          className="ZDlJU6tECg7ouG-b27gya _isfJqqskSRqop8TStvGA"
          data-testid="checkmark"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            className="_2GsLyQLHv8yNULHeXGdver"
            d="M200.655 513.229c-3.936-4.075-8.644-7.326-13.849-9.562s-10.804-3.413-16.47-3.465c-5.665-0.047-11.283 1.033-16.527 3.179-5.244 2.142-10.007 5.312-14.013 9.318s-7.174 8.768-9.32 14.012c-2.145 5.244-3.225 10.863-3.176 16.529s1.226 11.264 3.462 16.469c2.236 5.205 5.487 9.911 9.562 13.85l200.491 225.83c8.001 7.996 18.851 12.489 30.165 12.489s22.164-4.493 30.165-12.489l481.828-481.832c4.079-3.936 7.326-8.644 9.566-13.849 2.236-5.206 3.409-10.804 3.46-16.47s-1.028-11.283-3.174-16.527c-2.146-5.244-5.316-10.007-9.323-14.013-4.002-4.006-8.768-7.174-14.012-9.32s-10.863-3.225-16.525-3.176c-5.666 0.049-11.264 1.226-16.469 3.462-5.21 2.236-9.916 5.487-13.85 9.562l-451.667 451.666-170.325-195.665z"
          />
        </svg>
      </span>
    </span>
    <span
      className="label"
    >
      <p
        className="base regular base greyBlueDark"
      >
        Dont click me
      </p>
    </span>
  </label>
</div>
`;

exports[`renders each variation of checked, defaultChecked and indeterminate 1`] = `
<div
  className="checkbox"
>
  <label
    className="wrapper"
  >
    <span
      className="checkHolder"
    >
      <input
        aria-label="Foo"
        checked={true}
        className="input indeterminate"
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="checkBox"
      >
        <svg
          className="ZDlJU6tECg7ouG-b27gya _isfJqqskSRqop8TStvGA"
          data-testid="minus2"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            className="_2GsLyQLHv8yNULHeXGdver"
            d="M781.256 452.984h-535.558c-28.482 0-30.9 23.072-30.9 51.5s2.418 51.5 30.9 51.5h535.558c28.482 0 30.951-23.072 30.951-51.5s-2.47-51.5-30.951-51.5z"
          />
        </svg>
      </span>
    </span>
    <span
      className="label"
    >
      <p
        className="base regular base greyBlueDark"
      >
        Foo
      </p>
    </span>
  </label>
</div>
`;

exports[`renders each variation of checked, defaultChecked and indeterminate 2`] = `
<div
  className="checkbox"
>
  <label
    className="wrapper"
  >
    <span
      className="checkHolder"
    >
      <input
        aria-label="Foo"
        checked={false}
        className="input indeterminate"
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="checkBox"
      >
        <svg
          className="ZDlJU6tECg7ouG-b27gya _isfJqqskSRqop8TStvGA"
          data-testid="minus2"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            className="_2GsLyQLHv8yNULHeXGdver"
            d="M781.256 452.984h-535.558c-28.482 0-30.9 23.072-30.9 51.5s2.418 51.5 30.9 51.5h535.558c28.482 0 30.951-23.072 30.951-51.5s-2.47-51.5-30.951-51.5z"
          />
        </svg>
      </span>
    </span>
    <span
      className="label"
    >
      <p
        className="base regular base greyBlueDark"
      >
        Foo
      </p>
    </span>
  </label>
</div>
`;

exports[`renders each variation of checked, defaultChecked and indeterminate 3`] = `
<div
  className="checkbox"
>
  <label
    className="wrapper"
  >
    <span
      className="checkHolder"
    >
      <input
        aria-label="Foo"
        checked={true}
        className="input"
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="checkBox"
      >
        <svg
          className="ZDlJU6tECg7ouG-b27gya _isfJqqskSRqop8TStvGA"
          data-testid="checkmark"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            className="_2GsLyQLHv8yNULHeXGdver"
            d="M200.655 513.229c-3.936-4.075-8.644-7.326-13.849-9.562s-10.804-3.413-16.47-3.465c-5.665-0.047-11.283 1.033-16.527 3.179-5.244 2.142-10.007 5.312-14.013 9.318s-7.174 8.768-9.32 14.012c-2.145 5.244-3.225 10.863-3.176 16.529s1.226 11.264 3.462 16.469c2.236 5.205 5.487 9.911 9.562 13.85l200.491 225.83c8.001 7.996 18.851 12.489 30.165 12.489s22.164-4.493 30.165-12.489l481.828-481.832c4.079-3.936 7.326-8.644 9.566-13.849 2.236-5.206 3.409-10.804 3.46-16.47s-1.028-11.283-3.174-16.527c-2.146-5.244-5.316-10.007-9.323-14.013-4.002-4.006-8.768-7.174-14.012-9.32s-10.863-3.225-16.525-3.176c-5.666 0.049-11.264 1.226-16.469 3.462-5.21 2.236-9.916 5.487-13.85 9.562l-451.667 451.666-170.325-195.665z"
          />
        </svg>
      </span>
    </span>
    <span
      className="label"
    >
      <p
        className="base regular base greyBlueDark"
      >
        Foo
      </p>
    </span>
  </label>
</div>
`;

exports[`renders each variation of checked, defaultChecked and indeterminate 4`] = `
<div
  className="checkbox"
>
  <label
    className="wrapper"
  >
    <span
      className="checkHolder"
    >
      <input
        aria-label="Foo"
        checked={false}
        className="input"
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="checkBox"
      >
        <svg
          className="ZDlJU6tECg7ouG-b27gya _isfJqqskSRqop8TStvGA"
          data-testid="checkmark"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            className="_2GsLyQLHv8yNULHeXGdver"
            d="M200.655 513.229c-3.936-4.075-8.644-7.326-13.849-9.562s-10.804-3.413-16.47-3.465c-5.665-0.047-11.283 1.033-16.527 3.179-5.244 2.142-10.007 5.312-14.013 9.318s-7.174 8.768-9.32 14.012c-2.145 5.244-3.225 10.863-3.176 16.529s1.226 11.264 3.462 16.469c2.236 5.205 5.487 9.911 9.562 13.85l200.491 225.83c8.001 7.996 18.851 12.489 30.165 12.489s22.164-4.493 30.165-12.489l481.828-481.832c4.079-3.936 7.326-8.644 9.566-13.849 2.236-5.206 3.409-10.804 3.46-16.47s-1.028-11.283-3.174-16.527c-2.146-5.244-5.316-10.007-9.323-14.013-4.002-4.006-8.768-7.174-14.012-9.32s-10.863-3.225-16.525-3.176c-5.666 0.049-11.264 1.226-16.469 3.462-5.21 2.236-9.916 5.487-13.85 9.562l-451.667 451.666-170.325-195.665z"
          />
        </svg>
      </span>
    </span>
    <span
      className="label"
    >
      <p
        className="base regular base greyBlueDark"
      >
        Foo
      </p>
    </span>
  </label>
</div>
`;

exports[`renders each variation of checked, defaultChecked and indeterminate 5`] = `
<div
  className="checkbox"
>
  <label
    className="wrapper"
  >
    <span
      className="checkHolder"
    >
      <input
        aria-label="Foo"
        className="input indeterminate"
        defaultChecked={true}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="checkBox"
      >
        <svg
          className="ZDlJU6tECg7ouG-b27gya _isfJqqskSRqop8TStvGA"
          data-testid="minus2"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            className="_2GsLyQLHv8yNULHeXGdver"
            d="M781.256 452.984h-535.558c-28.482 0-30.9 23.072-30.9 51.5s2.418 51.5 30.9 51.5h535.558c28.482 0 30.951-23.072 30.951-51.5s-2.47-51.5-30.951-51.5z"
          />
        </svg>
      </span>
    </span>
    <span
      className="label"
    >
      <p
        className="base regular base greyBlueDark"
      >
        Foo
      </p>
    </span>
  </label>
</div>
`;

exports[`renders each variation of checked, defaultChecked and indeterminate 6`] = `
<div
  className="checkbox"
>
  <label
    className="wrapper"
  >
    <span
      className="checkHolder"
    >
      <input
        aria-label="Foo"
        className="input indeterminate"
        defaultChecked={false}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="checkBox"
      >
        <svg
          className="ZDlJU6tECg7ouG-b27gya _isfJqqskSRqop8TStvGA"
          data-testid="minus2"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            className="_2GsLyQLHv8yNULHeXGdver"
            d="M781.256 452.984h-535.558c-28.482 0-30.9 23.072-30.9 51.5s2.418 51.5 30.9 51.5h535.558c28.482 0 30.951-23.072 30.951-51.5s-2.47-51.5-30.951-51.5z"
          />
        </svg>
      </span>
    </span>
    <span
      className="label"
    >
      <p
        className="base regular base greyBlueDark"
      >
        Foo
      </p>
    </span>
  </label>
</div>
`;

exports[`renders each variation of checked, defaultChecked and indeterminate 7`] = `
<div
  className="checkbox"
>
  <label
    className="wrapper"
  >
    <span
      className="checkHolder"
    >
      <input
        aria-label="Foo"
        className="input"
        defaultChecked={true}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="checkBox"
      >
        <svg
          className="ZDlJU6tECg7ouG-b27gya _isfJqqskSRqop8TStvGA"
          data-testid="checkmark"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            className="_2GsLyQLHv8yNULHeXGdver"
            d="M200.655 513.229c-3.936-4.075-8.644-7.326-13.849-9.562s-10.804-3.413-16.47-3.465c-5.665-0.047-11.283 1.033-16.527 3.179-5.244 2.142-10.007 5.312-14.013 9.318s-7.174 8.768-9.32 14.012c-2.145 5.244-3.225 10.863-3.176 16.529s1.226 11.264 3.462 16.469c2.236 5.205 5.487 9.911 9.562 13.85l200.491 225.83c8.001 7.996 18.851 12.489 30.165 12.489s22.164-4.493 30.165-12.489l481.828-481.832c4.079-3.936 7.326-8.644 9.566-13.849 2.236-5.206 3.409-10.804 3.46-16.47s-1.028-11.283-3.174-16.527c-2.146-5.244-5.316-10.007-9.323-14.013-4.002-4.006-8.768-7.174-14.012-9.32s-10.863-3.225-16.525-3.176c-5.666 0.049-11.264 1.226-16.469 3.462-5.21 2.236-9.916 5.487-13.85 9.562l-451.667 451.666-170.325-195.665z"
          />
        </svg>
      </span>
    </span>
    <span
      className="label"
    >
      <p
        className="base regular base greyBlueDark"
      >
        Foo
      </p>
    </span>
  </label>
</div>
`;

exports[`renders each variation of checked, defaultChecked and indeterminate 8`] = `
<div
  className="checkbox"
>
  <label
    className="wrapper"
  >
    <span
      className="checkHolder"
    >
      <input
        aria-label="Foo"
        className="input"
        defaultChecked={false}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="checkBox"
      >
        <svg
          className="ZDlJU6tECg7ouG-b27gya _isfJqqskSRqop8TStvGA"
          data-testid="checkmark"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            className="_2GsLyQLHv8yNULHeXGdver"
            d="M200.655 513.229c-3.936-4.075-8.644-7.326-13.849-9.562s-10.804-3.413-16.47-3.465c-5.665-0.047-11.283 1.033-16.527 3.179-5.244 2.142-10.007 5.312-14.013 9.318s-7.174 8.768-9.32 14.012c-2.145 5.244-3.225 10.863-3.176 16.529s1.226 11.264 3.462 16.469c2.236 5.205 5.487 9.911 9.562 13.85l200.491 225.83c8.001 7.996 18.851 12.489 30.165 12.489s22.164-4.493 30.165-12.489l481.828-481.832c4.079-3.936 7.326-8.644 9.566-13.849 2.236-5.206 3.409-10.804 3.46-16.47s-1.028-11.283-3.174-16.527c-2.146-5.244-5.316-10.007-9.323-14.013-4.002-4.006-8.768-7.174-14.012-9.32s-10.863-3.225-16.525-3.176c-5.666 0.049-11.264 1.226-16.469 3.462-5.21 2.236-9.916 5.487-13.85 9.562l-451.667 451.666-170.325-195.665z"
          />
        </svg>
      </span>
    </span>
    <span
      className="label"
    >
      <p
        className="base regular base greyBlueDark"
      >
        Foo
      </p>
    </span>
  </label>
</div>
`;
